base/src/main/java/com/smartdevicelink/managers/screen/TextAndGraphicUpdateOperation.java

```diff
@@ -452,42 +501,59 @@ private Show setBlankTextFields(Show newShow) {
         return newShow;
     }
 
+    Show assembleLayout(Show show) {
+        if (!(internalInterface.get().getSdlMsgVersion().getMajorVersion() >= 6) || !shouldUpdateTemplateConfig()) {
+            return show;
+        }
+        show.setTemplateConfiguration(updatedState.getTemplateConfiguration());
+        return show;
+    }
+
+    private void updateCurrentScreenDataFromSetDisplayLayout(SetDisplayLayout setDisplayLayout) {
+        currentScreenData.setTemplateConfiguration(new TemplateConfiguration().setTemplate(setDisplayLayout.getDisplayLayout()).setDayColorScheme(setDisplayLayout.getDayColorScheme()).setNightColorScheme(setDisplayLayout.getNightColorScheme()));
+        if(currentScreenDataUpdateListener != null){
+            currentScreenDataUpdateListener.onUpdate(currentScreenData);
+        }
+    }
+
     private void updateCurrentScreenDataFromShow(Show show) {
         if (show == null) {
             DebugTool.logError(TAG, "can not updateCurrentScreenDataFromShow from null show");
             return;
         }
-
-        // If the items are null, they were not updated, so we can't just set it directly
         if (show.getMainField1() != null) {
-            currentScreenData.setMainField1(show.getMainField1());
+            currentScreenData.setTextField1(updatedState.getTextField1());
```

Comment:
we can not set an sdlArtWork object from a show, so if at this point the show has a value, it was updated on the screen, so we check the show, then transfer the value form the updateState to currentScreenData. We did this because in TextAndGraphicsManager if an update fails we want to reset the values to what is displaying on screen, so when a user does sdlManager.getTextField1, it's getting the value that's on screen. I can change the name if you want to updateCurrenScreenDataFromUpdateState, but we still have to take in a show to see if what values are set in it.